rust/operator-binary/src/controller.rs

```diff
@@ -354,13 +421,45 @@ fn build_metastore_rolegroup_statefulset(
     hive: &HiveCluster,
     rolegroup_ref: &RoleGroupRef<HiveCluster>,
     metastore_config: &HashMap<PropertyNameKind, BTreeMap<String, String>>,
+    s3_connection: Option<&S3ConnectionSpec>,
 ) -> Result<StatefulSet> {
     let mut db_type: Option<DbType> = None;
     let mut container_builder = ContainerBuilder::new(APP_NAME);
 
     for (property_name_kind, config) in metastore_config {
         match property_name_kind {
             PropertyNameKind::Env => {
+                if let Some(S3ConnectionSpec {
+                    secret_class: Some(secret_name),
+                    ..
+                }) = s3_connection
+                {
+                    container_builder.add_env_var_from_source(
```

Comment:
What do you think about using `add_env_var_from_secret` from https://github.com/stackabletech/operator-rs/blob/d8687081325924ee68069f39b873f1f25bf0ed09/src/builder/pod/container.rs#L86?

Reply:
yep way nice, did not see that!